airflow/providers/amazon/aws/sensors/cloud_formation.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains sensors for AWS CloudFormation.
+"""
+from botocore.exceptions import ClientError
+
+from airflow.providers.amazon.aws.hooks.cloud_formation import AWSCloudFormationHook
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class BaseCloudFormationSensor(BaseSensorOperator):
+    """
+    Waits for a stack operation to complete on AWS CloudFormation.
+
+    :param stack_name: The name of the stack to wait for (templated)
+    :type stack_name: str
+    :param aws_conn_id: ID of the Airflow connection where credentials and extra configuration are
+        stored
+    :type aws_conn_id: str
+    :param poke_interval: Time in seconds that the job should wait between each try
+    :type poke_interval: int
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 stack_name,
+                 complete_status,
+                 in_progress_status,
+                 aws_conn_id='aws_default',
+                 poke_interval=60 * 3,
+                 *args,
+                 **kwargs):
+        super().__init__(poke_interval=poke_interval, *args, **kwargs)
+        self.aws_conn_id = aws_conn_id
+        self.stack_name = stack_name
+        self.complete_status = complete_status
+        self.in_progress_status = in_progress_status
+        self.hook = None
+
+    def poke(self, context):
+        """
+        Checks for existence of the stack in AWS CloudFormation.
+        """
+        cloudformation = self.get_hook().get_conn()
```

Comment:
```suggestion
        hook = AWSCloudFormationHook(aws_conn_id=self.aws_conn_id)
        cloudformation = hook.get_conn()
```

No need to use a hook as instance attribute and it's clearer and simpler

Reply:
This will create a new hook every time poke is called, is that something that is desired?